### ISSUE TITLE
what's with Xavier ? should we are clone the repo and making further ?

### ISSUE THREAD

**Issue by blackcrack** [DjazzLab/ocdownloader#58]
Hi,

this is a issue, OCdownloader it is a really well addon for owncloud and nextcloud,
we should have anybody who developing this further, because things downloading 
in the owncloud with a downloader it's a really nice thing !
can any body maybe developing this nice piece  of addon further ?

best regards 
Blacky

**Comment by slavoutich** [comment]
Would be an extremely good idea, as for me. App works for me now (except torrent restart after a reboot), but I think it is temporarily without a maintenance.

**Comment by DJCrashdummy** [comment]
well... i'm a little bit worried that something happened to @DjazzLab !?!
at the only left online-appearance (beside github), which is twitter, [the last update was from february 2016](https://twitter.com/DjazzLab/status/697347571086446592) with the preannouncement of ocDownloader v2.0, similar to https://github.com/DjazzLab/ocdownloader/issues/45#issuecomment-198667458.

i regularly miss this app, because especially for bigger files it is really tedious downloading it and afterwards upload it (with usual crappy upload speed) to my cloud. ...so the time for this task gets multiplied! :-(

maybe this useful app can be forked and the further developed done under the umbrella of @nextcloud, if they are interested @jospoortvliet? --> hopefully you are the right person to talk to, if not sorry for mentioning.
since i sadly don't know anything similar for nextcloud and it is still rated pretty good at the [old owncloud app-store](https://apps.owncloud.com/content/show.php/ocDownloader+%28NG%29?content=169974) it would be a really nice addition for the nextcloud app-store.

**Comment by blackcrack** [comment]
i guess, something has happens with Xavier, what we can not change.. and it is maybe, like he have posted the pictures, have make a out time or some.. something other want i am not imagine.. but well, if he not 
obtainable or something it is..  and easy not reachable on other way.. should someone make easy a fork because this down loader for xxx-cloud and i guess would a really thing for have further a working a 
cloud-downloader who it is maintained . \*fingercrossing\*

**Comment by e-alfred** [comment]
Yes, forking would be wise at this point, probably. A lot of Owncloud/Nextcloud apps get abandoned by their developers after some time, so this is no exception. A fork that addresses the biggest problems would be a good place to start at this point.

**Comment by DJCrashdummy** [comment]
@e-alfred you are right, but i want/have to point one thing out:
if ocDownloader should be added to the nextcloud-appstore sometimes, there will be a license-issue we have to solve, since only AGPLv3, MPL 2.0 or Apache License 2.0 are allowed and its [license was changed from AGPL3 to CC BY-SA](https://github.com/DjazzLab/ocdownloader/commit/7b875dd3301734041d8ec263f2eb1e56f30340b3).

- i'm not a license-expert, so either they are compatible and we can change it back, or to another accepted license, which i doubt...
- or it may be better to start the fork from the last version under AGPL which wouldn't be a big drawback, since after the change the only [contributions were from Loki3000](https://github.com/DjazzLab/ocdownloader/commits?author=Loki3000).

**Comment by Nibbels** [comment]
Hello there :)

I personally wouldnt have the time to look into more developing. But I really like the app.
Why did I do this? I did implement Link-Buttons to the "News-Feed" app as well. Thus I can download those audio-feeds and they get pushed to my phone without any effort. (...)
So I only needed a strong http-downloader.

One problem is this *.sh-File I guess. It has to have special chmod / chown. I think I read that this requirement is something that nextcloud/owncloud dislikes.

**Comment by jospoortvliet** [comment]
I can imagine that that is a security issue, yeah, plus it makes installation harder (or impossible, on some systems). A less invasive solution would be very welcome. Besides that, if somebody wants to maintain it, we'd love to have it and it could be moved to the Nextcloud project. However the license is indeed weird, I am not sure if CC BY-SA is legally possible with the AGPLv3 and of course we don't want to encourage license violations of mixing incompatible code from our app store.

**Comment by e-alfred** [comment]
I found this on the topic of CC-BY-SA and GPL compatibility: https://wiki.creativecommons.org/wiki/ShareAlike_compatibility:_GPLv3#Considerations_for_adapters_applying_the_GPLv3

https://wiki.creativecommons.org/wiki/ShareAlike_compatibility

This app is CC-BY-SA Version 3 or later, so it could apply here.

A better way though would be to take the last AGPL licensed version and kindly ask @Nibbels and @Loki3000 to reapply their patches to a fork under the AGPL license.

**Comment by DJCrashdummy** [comment]
@e-alfred again, just a small hint preventively: GPL != AGPL and thus GPL isn't accepted by the nextcloud-appstore either.

so it is not just a _better way_, it is the only way if ocDownloader should be added to the nextcloud-store.

**Comment by e-alfred** [comment]
@DJCrashdummy To get around the license quagmire, I created a fork with the last AGPL version: https://github.com/e-alfred/ocdownloader

To complete this task, I would like to kindly ask @Nibbels and @Loki3000 to resupply their patches licensed as AGPL V3 or later.

From that point onward, Nextcloud app store compatibility should be archived and it could be put under the umbrella of Nextcloud if somebody would like to work on it. Renaming it to something like ncDownloader would probably also be a good idea.

**Comment by jospoortvliet** [comment]
that is awesome, @e-alfred !!!

**Comment by Nibbels** [comment]
Nice :) 
Will you rename it to ncDownloader or will it stay as it is?
Of course I will start implementing my fix when this is question cleared out. 

This project might be alot of fun is someone has more experience in "app best practices for newer nextcloud versions" than I have. I had to do a lot of reading to find out what routines should be used and which are marked as depreached. (reading user id, reading path, ...)
That is something I cannot do within my freetime rightnow. Testing and debugging: yes.

**Comment by DJCrashdummy** [comment]
@e-alfred thank you for this good start!

i'm voting for simply _**Downloader**_, since everybody should know which cloud he is using... or _**cloudDownloader**_ in lack of creativity on my side. ;-) 

i saw @yp05327 worked recently on a nexcloud-adaption, maybe he is also so kind and merges his changes to this new project?

**Comment by Nibbels** [comment]
hmm
https://www.google.de/search?q=cloudgrabber&oq=cloudgrabber
https://www.google.de/search?q=cloudDownloader

https://www.google.de/search?q=ncdownloader
"NCDownloader is an ad-supported product that installs various toolbars during the installation and setup process. Users are not given the option to not include"
bad choice..

**Comment by yp05327** [comment]
@DJCrashdummy 
I'm using this addon on my own nextcloud server now, but what I have done is just made it run xD. I don't know how to develop addons for nextcloud.

About my fork: 
1、I haven't do any changes to this addon's code, the code has no problem, just lose a necessary js file. This file can find in my fork's readme.md
2、some times there's a bug, maybe a bug I think, I can't see what has been downloaded. This is because nextcloud is not always refreshing the changes of files, so when a new file move to a user's data floder by some 'illegal' ways, you can not see the file. So I changed nextcloud's config, set it always refresh the changes of files.
3、necessary download engine like ARIA2/YouTube-DL and so on, I have just tested ARIA2, don't know the performance of YouTube-DL, also other unknow download engines. May be can change a better one?(Joke, I have no idea of linux download engine).

Suggestion: (If you want to add new codes or changes.)
I think the user page is too complex. When I have downloaded a file and want to remove the history information, I need to remove the history info in 'completes' page, and also the history info in 'removed' page.It is difficult to use I think. But it is easy to the developers to design lol.

This is a usefull addon I think, but unfortunately the owner doesn't commit any more. If new project created, please tell me the link or name xD, thanks.

**Comment by Nibbels** [comment]
hello @yp05327, my Pull-Request would fix this file refresh issue at point 2.
And I can confirm your point with the "completes" and "removed" history.

**Comment by e-alfred** [comment]
@Nibbels Maybe you can push the pull request to my repository that took the last AGPL version so we can go forward with a Nextcloud store compatible codebase from a license point of view?

**Comment by Nibbels** [comment]
@e-alfred
I downloaded your version, changed it offline and patched it over.
Then started the pull request.
https://github.com/e-alfred/ocdownloader/pull/1

**Comment by blackcrack** [comment]
which version should we now use as headversion of ocdownloader on which repo is the future "headversion" of OCdownloader ? on this repo shuld we working together.. 
And @Nibbels , all where wrote here, fellow this tread, like i guss ;) 

best regards
Blacky

**Comment by DJCrashdummy** [comment]
@e-alfred you are completely right, but i just started thinking loud, since nibbels brought up the topic.

@blackcrack well... to make it short: "time will tell" - maybe in the future there will be an other fork of the latest original ocDownloader with better features, fewer bugs, etc. which will take over the lead or maybe the original will continue, but (for now) e-alfreds fork is the best choice, since all forks don't differentiate much from another but his fork can be added to the nexcloud-appstore without any license-issues.

**Comment by blackcrack** [comment]
Hi @DJCrashdummy ,

it was to abrupt where have Xavier nothing done any more.. i guess a continuing of the orginal Repo.. i guss it is no more, because Xavier had say anything in this tread if Xavier would be already in the net,
so it is  the @e-alfred  the best choice to working under e-alfred repo further.. i guess so feel we are all
and can only hold up the terriffic/Magnificent/great idea and work from Xavier in future too to have for all the best possible in the Clouds in the world from Xavier's founding.

so, e-alfred  .. and thank you very much e-alfred 👍 

\*Small tears in the eye\*

best regards
Blacky

**Comment by blackcrack** [comment]
the work is done, in memory to Xavier

thank you very much to all ! :)

best regards
Blacky

thank you Nibbles 👍

**Comment by Nibbels** [comment]
I found this:
https://twitter.com/Djazzlab

I guess somewhere in this universe is a ocdownloader 2.0 offline available ^^

**Comment by blackcrack** [comment]
well, sgc-univ.net  it is offline ..

do you be in facebook ? (i am not)
https://www.facebook.com/Djazzlab/posts/807652792646305
found in google:
```
SGC-Univ.Net - #ownCloud #ocDownloader reborn now using ...
https://www.facebook.com/Djazzlab/posts/807652792646305 - Diese Seite übersetzen
Hi guys ! Two years ago, I developed an application for ownCloud called ocDownloader. This application allowed you to download files from the Internet directly ...
```
and on 

```
SGC-Univ.Net | Facebook
https://pt-br.facebook.com/Djazzlab/ - Diese Seite übersetzen
SGC-Univ.Net. 19 curtidas · 1 falando sobre isso. Computer things I'm working on every days … and all the things that make my geek life easier :)...
```
with http://lmgtfy.com/?q=+sgc-univ.net
and http://lmgtfy.com/?q=+Djazzlab
all more as a year ago..

i bet/guess he has known he is sick or something..
what do you think why he have bring in so many energy 
in so short time in his project ...
and the picture was from a better Day's :\ hmm 

this was the last message what i have read .. and then.. bumm he was gone..
https://gitter.im/DjazzLab/ocdownloader
so he has -=>plan<=- a app for handy(app) so it is anything shortly happend..
\*sniff\*

best regards
blacky

**Comment by e-alfred** [comment]
His Facebook page is gone too. He announced ocDownloader 2.0 shortly before he vanished into thin air.

If anybody wants to help out, here is the AGPL licensed version with all the fixes from @Nibbels and me. This version can be supplied to the Nextcloud store as soon as it is good enough for that.

**Comment by jospoortvliet** [comment]
I dunno who did it but now the app is on the app store, so I created a category on the forums to discuss it 🎉 🎈 
https://help.nextcloud.com/c/apps/ocdownloader

**Comment by e-alfred** [comment]
@jospoortvliet Well if you look at the website and issue tracker given (and who the CSR uploaded) then I think you know who put it there.  :) Sadly the app store somehow doesn't evaluate the author tags in the info.xml file so it shows up as "anonymous".

**Comment by jospoortvliet** [comment]
@e-alfred aw you're right, you're the hero 😈  rocking! @BernhardPosselt what's going wrong there?

**Comment by BernhardPosselt** [comment]
The author tags are atm only exported in the API: Anonymous is chosen if the uploader profiles has no firstname and/or lastname set.

**Comment by BernhardPosselt** [comment]
See https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/core/templates/app/detail.html#L68

**Comment by jospoortvliet** [comment]
lol

documentation by code 📦

**Comment by BernhardPosselt** [comment]
Here's docs by docs ;) http://nextcloudappstore.readthedocs.io/en/latest/developer.html#info-xml

ctrl f author

**Comment by blackcrack** [comment]
the letters and docu's of the Programmers be mostly in /the code ;) @jospoortvliet ;)
normal thinking is more hard because you must bring you self in a lower level, 
as User or something... it's something like you speak in Latin and must start again 
to speak in you mother lang if you live in Rome and you have long no more speak
with your motherlang, this is ~ the same, so it's more easy, you speak/write in code ..
if you many programming and be mostly in a silence room .. if i am not wrong.. :-] 
best regards
Blacky
